### PR TITLE
Validation issues + Storybook

### LINKED ISSUE #19: Subscribe: Linkedin URL validation not working
I entered https://linkedin.com/in/stevejalim but the field remained red and the subscribe form unsubmittable even though the help text said "https://linkedin.com/in/stevejalim valid"

### PR DESCRIPTION
**Pull request by silviopaganini**
fix #19 and adds storybook for easy component debug

### CODE REVIEW

**Comment by silviopaganini** [comment]
last commit fix #17

**Comment by silviopaganini** [comment]
last commit fixes #12 so @andrevenancio is not offended

File: src/components/form/validation.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const Validation = {
   validName: (value) => {
-    const re = /[\w]+[ ][\w]+/;
+    const re = /^[a-zA-ZàáâäãåąčćęèéêëėįìíîïłńòóôöõøùúûüųūÿýżźñçčšžÀÁÂÄÃÅĄĆČĖĘÈÉÊËÌÍÎÏĮŁŃÒÓÔÖÕØÙÚÛÜŲŪŸÝŻŹÑßÇŒÆČŠŽ∂ð ,.'-]+[ ][a-zA-ZàáâäãåąčćęèéêëėįìíîïłńòóôöõøùúûüųūÿýżźñçčšžÀÁÂÄÃÅĄĆČĖĘÈÉÊËÌÍÎÏĮŁŃÒÓÔÖÕØÙÚÛÜŲŪŸÝŻŹÑßÇŒÆČŠŽ∂ð ,.'-]+$/;
```
Comment (tevla) [inline src/components/form/validation.js]: if we have to use so many special characters in the regexp, why not just simply remove the validation in the name all together? Do we need to validate someone's name?

File: src/components/form/validation.js [hunk 2]
```diff
@@ -19,20 +19,21 @@ const Validation = {
...
   },
   validEmail: (value) => {
     // found somewhere online
-    const re = /^([a-zA-Z0-9_\-.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([a-zA-Z0-9-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$/;
+    // eslint-disable-next-line
+    const re = /^((([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+(\.([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+)*)|((\x22)((((\x20|\x09)*(\x0d\x0a))?(\x20|\x09)+)?(([\x01-\x08\x0b\x0c\x0e-\x1f\x7f]|\x21|[\x23-\x5b]|[\x5d-\x7e]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(\\([\x01-\x09\x0b\x0c\x0d-\x7f]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]))))*(((\x20|\x09)*(\x0d\x0a))?(\x20|\x09)+)?(\x22)))@((([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.)+(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.?$/;
```
Comment (tevla) [inline src/components/form/validation.js]: this works, maybe we can integrate later on an email verification so we can remove all of these tests too

File: src/components/form/validation.js (hunk 1 above)
Comment (silviopaganini) [reply src/components/form/validation.js]: hmmm it's nice to have [First][ ][Last] names validated I guess

File: src/components/form/validation.js (hunk 2 above)
Comment (silviopaganini) [reply src/components/form/validation.js]: how do you mean?

File: src/components/form/Form.stories.jsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import MemberFields from './MemberFields';
+
+storiesOf('Body', module)
+  .add('default', () => (
+    <MemberFields />
```
Comment (grudelsud) [inline src/components/form/Form.stories.jsx]: no idea what's this for, but looks cool. @silviopaganini maybe the other fieldset for advertisers should be in storybook as well? how is that used? maybe the form with the radio button as well? or the member's card switching to form? 😂

File: src/components/form/validation.js [hunk 4]
```diff
@@ -19,20 +19,21 @@ const Validation = {
     return value && counter < 1 ? undefined : msg;
   },
   validGithub: (value) => {
-    const re = /https:\/\/github\.com\/[A-z0-9_-]+\/?/;
+    const re = /https:\/\/(www\.)?github\.com\/[A-z0-9_-]+\/?/;
```
Comment (grudelsud) [inline src/components/form/validation.js]: both github and twitter have redirects so their urls never have www, which is the reason why I didn't put them in the first place.

File: .gitignore [hunk 5]
```diff
@@ -1,5 +1,7 @@
 # See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
 
+storybook-static/
```
Comment (grudelsud) [inline .gitignore]: guess this needed to be in? after the build I found a folder full of stuff

File: src/components/form/validation.js (hunk 2 above)
Comment (tevla) [reply src/components/form/validation.js]: I found that the easiest way to test if an email is valid is to actually ask the user to validate their email address.

File: src/components/form/Form.stories.jsx (hunk 3 above)
Comment (silviopaganini) [reply src/components/form/Form.stories.jsx]: That’s just to quickly debug UI, rendering it separately from the rest of the build. When we need test/develop the rest we’ll create them

File: src/components/form/validation.js (hunk 4 above)
Comment (silviopaganini) [reply src/components/form/validation.js]: Well, let their servers do the job then. www. Is still valid

File: src/components/form/validation.js (hunk 2 above)
Comment (silviopaganini) [reply src/components/form/validation.js]: Hmmm we don’t have that service yet, to dispatch emails, let’s keep it for now, then once we have the email server up we can remove it

**Comment by silviopaganini** [comment]
Merge?

**Pull request merged**